libmariadb/mariadb_lib.c

```diff
@@ -2537,7 +2537,11 @@ mysql_stat(MYSQL *mysql)
 int STDCALL
 mysql_ping(MYSQL *mysql)
 {
-  return ma_simple_command(mysql, COM_PING,0,0,0,0);
+  int rc;
+  rc= ma_simple_command(mysql, COM_PING, 0, 0, 0, 0);
+  if (rc && mysql->options.reconnect)
+    rc= ma_simple_command(mysql, COM_PING, 0, 0, 0, 0);
+  return rc;
 }
```

Comment:
Hmm, perhaps you can reconnect only on certain errors? Like, no need to reconnect if the first ping got "access denied"

Reply:
Ok, maybe only test if error code was CR_SERVER_LOST ?! (like it was before)

Reply:
@9E0R9 yes. Why did you remove that (your commit had no explanation for removal)?

@vaintroub yes, when using one-time passwords (google authenticator, rsa securid, etc) one cannot reconnect using old credentials.

Reply:
I think we should check these error codes:

CR_SERVER_GONE_ERROR (2006)
CR_SERVER_LOST (2013)